File: java/src/sdk/dataobjects/paypal/payflow/Address.java
/**
 * This program is free software: you can redistribute it and/or modify
 *    it under the terms of the GNU General Public License as published by
 *    the Free Software Foundation, either version 3 of the License, or
 *    (at your option) any later version.
 *
 *   This program is distributed in the hope that it will be useful,
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *   GNU General Public License for more details.
 *
 *   You should have received a copy of the GNU General Public License
 *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */package paypal.payflow;

/// <summary>
/// Abstract class to hold the Address information.
/// </summary>

/// <remarks>This class can be extended to create a new address type.</remarks>
public abstract class Address extends BaseRequestDataObject {

    private String addressStreet;
    private String addressStreet2;
    private String city;
    private String state;
    private String zip;
    private String country;
    private String phoneNum;
    private String phone2;
    private String email;
    private String fax;
    private String firstName;
    private String middleName;
    private String lastName;


    /// <summary>
    /// Constructor for Address.
    /// </summary>
    /// <remarks>Abstract class. Instance cannot be created directly.</remarks>
    protected Address() {
    }

    protected String getAddressStreet() {
        return addressStreet;
    }

    protected void setAddressStreet(String addressStreet) {
        this.addressStreet = addressStreet;
    }

    protected String getAddressStreet2() {
        return addressStreet2;
    }

    protected void setAddressStreet2(String addressStreet2) {
        this.addressStreet2 = addressStreet2;
    }

    protected String getAddressCity() {
        return city;
    }

    protected void setAddressCity(String city) {
        this.city = city;
    }

    protected String getAddressCountry() {
        return country;
    }

    protected void setAddressCountry(String country) {
        this.country = country;
    }

    protected String getAddressEmail() {
        return email;
    }

    protected void setAddressEmail(String email) {
        this.email = email;
    }

    protected String getAddressFax() {
        return fax;
    }

    protected void setAddressFax(String fax) {
        this.fax = fax;
    }

    protected String getAddressFirstName() {
        return firstName;
    }

    protected void setAddressFirstName(String firstName) {
        this.firstName = firstName;
    }

    protected String getAddressLastName() {
        return lastName;
    }

    protected void setAddressLastName(String lastName) {
        this.lastName = lastName;
    }

    protected String getAddressMiddleName() {
        return middleName;
    }

    protected void setAddressMiddleName(String middleName) {
        this.middleName = middleName;
    }

    protected String getAddressPhone2() {
        return phone2;
    }

    protected void setAddressPhone2(String phone2) {
        this.phone2 = phone2;
    }

    protected String getAddressPhone() {
        return phoneNum;
    }

    protected void setAddressPhone(String phoneNum) {
        this.phoneNum = phoneNum;
    }

    protected String getAddressState() {
        return state;
    }

    protected void setAddressState(String state) {
        this.state = state;
    }

    protected String getAddressZip() {
        return zip;
    }

    protected void setAddressZip(String zip) {
        this.zip = zip;
    }

}

File: dotNET/PayflowProSDK/PayPal/Payments/DataObjects/Address.cs
#region "Imports"

using System;

#endregion

namespace PayPal.Payments.DataObjects
{
	/// <summary>
	/// Abstract class to hold the
	/// Address information.
	/// </summary>
	/// <remarks>This class can be extended to create a new address type.</remarks>
	public abstract class Address : BaseRequestDataObject
	{
		#region "Member Variables"

		/// <summary>
		/// Holds Street
		/// </summary>
		private String mStreet;

		/// <summary>
		/// Holds Street2
		/// </summary>
		private String mStreet2;

		/// <summary>
		/// Holds City
		/// </summary>
		private String mCity;

		/// <summary>
		/// Holds State
		/// </summary>
		private String mState;

		/// <summary>
		/// Holds Zip
		/// </summary>
		private String mZip;

		/// <summary>
		/// Holds Country
		/// </summary>
		private String mCountry;

		/// <summary>
		/// Holds Phonenum
		/// </summary>
		private String mPhone;

		/// <summary>
		/// Holds Phone2
		/// </summary>
		private String mPhone2;

		/// <summary>
		/// Holds Email
		/// </summary>
		private String mEmail;

		/// <summary>
		/// Holds Fax
		/// </summary>
		private String mFax;

		/// <summary>
		/// Holds First Name
		/// </summary>
		private String mFirstName;

		/// <summary>
		/// Holds Middle Name
		/// </summary>
		private String mMiddleName;

		/// <summary>
		/// Holds Last Name
		/// </summary>
		private String mLastName;

		#endregion

		#region "Properties"

		/// <summary>
		/// Gets, Sets Street
		/// </summary>
		internal String AddressStreet
		{
			get { return mStreet; }
			set { mStreet = value; }
		}

		/// <summary>
		/// Gets, Sets Street2
		/// </summary>
		internal String AddressStreet2
		{
			get { return mStreet2; }
			set { mStreet2 = value; }
		}

		/// <summary>
		/// Gets, Sets City
		/// </summary>
		internal String AddressCity
		{
			get { return mCity; }
			set { mCity = value; }
		}

		/// <summary>
		/// Gets, Sets State
		/// </summary>
		internal String AddressState
		{
			get { return mState; }
			set { mState = value; }
		}

		/// <summary>
		/// Gets, Sets Zip
		/// </summary>
		internal String AddressZip
		{
			get { return mZip; }
			set { mZip = value; }
		}

		/// <summary>
		/// Gets, Sets Country
		/// </summary>
		internal String AddressCountry
		{
			get { return mCountry; }
			set { mCountry = value; }
		}

		/// <summary>
		/// Gets, Sets Phonenum
		/// </summary>
		internal String AddressPhone
		{
			get { return mPhone; }
			set { mPhone = value; }
		}

		/// <summary>
		/// Gets, Sets Phone2
		/// </summary>
		internal String AddressPhone2
		{
			get { return mPhone2; }
			set { mPhone2 = value; }
		}

		/// <summary>
		/// Gets, Sets Email
		/// </summary>
		internal String AddressEmail
		{
			get { return mEmail; }
			set { mEmail = value; }
		}

		/// <summary>
		/// Gets, Sets Fax
		/// </summary>
		internal String AddressFax
		{
			get { return mFax; }
			set { mFax = value; }
		}

		/// <summary>
		/// Gets, Sets Firstname
		/// </summary>
		internal String AddressFirstName
		{
			get { return mFirstName; }
			set { mFirstName = value; }
		}

		/// <summary>
		/// Gets, Sets Middlename
		/// </summary>
		internal String AddressMiddleName
		{
			get { return mMiddleName; }
			set { mMiddleName = value; }
		}

		/// <summary>
		/// Gets, Sets Lastname
		/// </summary>
		internal String AddressLastName
		{
			get { return mLastName; }
			set { mLastName = value; }
		}

		#endregion

		#region "Constructors"

		/// <summary>
		/// Constructor for Address.
		/// </summary>
		/// <remarks>Abstract class. Instance cannot be created directly.</remarks>
		protected Address()
		{
		}

		#endregion
	}

}